app/services/foreman/access_control.rb

```diff
@@ -29,6 +29,21 @@ def permissions
         @permissions
       end
 
+      def permissions_for_controller_action(controller_action)
+        controller_action = path_hash_to_string(controller_action) if controller_action.is_a?(Hash)
```

Comment:
We can't use `path_to_authenticate`, this is different context with no notion of controller instance. Also `path_to_authenticate` means something different, that's relative url that can contain id. We need this "$controller/$action" string since this is how it's defined in app/services/foreman/access_permissions.rb.

You're right that currently Foreman doesn't use this method with hash input but I wanted to introduce some API for AccessControl class that allows you to find permission based on this combination. I find it useful to be able to specify the string combination without creating the hash since you already might have such string from request.path.